### PR TITLE
Soback 107 smurf implementation

### PR DESCRIPTION
**Pull request by jorntx**
Smurf functionality added to toolbox. 

Egense added as co-reviewer to test out that all is good in Tomcat as this PR tampers with routing.

### CODE REVIEW

**Review by thomasegense — CHANGES_REQUESTED** [review]
Needs to stay in the toolbox at the cost of navigation.

**Review by thomasegense — APPROVED** [review]
Working inside toolbox. OK

File: src/js/src/components/ngrams/chartsCore/chartConfigs/chartOptionsCore.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        }
+      }]
+    }
+  },
+
+  /**
+  * Generate random color for lines
+  * Kudos - https://stackoverflow.com/questions/10014271/generate-random-color-distinguishable-to-humans 
+  */
+  getChartLineColor(salt){
```
Comment (jesperlauridsen) [inline src/js/src/components/ngrams/chartsCore/chartConfigs/chartOptionsCore.js:117]: Very cool!

File: src/js/src/components/ngrams/chartsCore/chartEngines/LineChart.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    },
+    options: {
+      type: Object,
+      default: null
+    }
+  },
+  mounted () {
+    // this.chartData is created in the mixin.
+    // If you want to pass options please create a local options object
+    this.renderChart(this.chartData,  this.options)
```
Comment (jesperlauridsen) [inline src/js/src/components/ngrams/chartsCore/chartEngines/LineChart.js:21]: This seems funky to me somehow. You never really create a new chart (like let chart = new Chart()) - but it somehow works with this vue-chartjs magic. Very funky, imho

File: src/js/src/store/modules/ngram.store.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
+import { requestService } from '../../services/RequestService'
+
+
+// Global ngram state.
+const initialState = () => ({
+  query: '',
+  // cache for current query in case server rejects it
+  attemptedQuery:'',
```
Comment (jesperlauridsen) [inline src/js/src/store/modules/ngram.store.js:8]: You don't use this attemptedQuery anywhere? At least I can't seem to find it...

File: src/js/src/components/ngrams/netarchive/SearchBox.vue [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+ 
+  data () {
+    return {    
+        searchQuery:''
+    }
+  },
+  
+  computed: {
+    ...mapState({
+     query: state => state.Ngram.query,
```
Comment (jesperlauridsen) [inline src/js/src/components/ngrams/netarchive/SearchBox.vue:37]: You also don't use the query from the store here it seems. You don't have the method for updating it set in your methods picked from the Ngram store in mapActions.

File: src/js/src/components/ngrams/chartsCore/chartEngines/LineChart.js (hunk 2 above)
Comment (jorntx) [reply src/js/src/components/ngrams/chartsCore/chartEngines/LineChart.js:21]: Very valid point and one of the drawbacks of using wrappers:(

I think we perhaps should have a future discussion about the usage of wrappers vs 'the raw product code'. 

The logic here is that the LineChart component is an instance of ChartJs hence the _this.renderChart_

Comment (jesperlauridsen) [reply src/js/src/components/ngrams/chartsCore/chartEngines/LineChart.js:21]: But isn't one of the drawbacks here, that you can't get the chart object (which you seem to need in your onClick function that's causing you problems?

File: src/js/src/store/modules/ngram.store.js (hunk 3 above)
Comment (jorntx) [reply src/js/src/store/modules/ngram.store.js:8]: Yes, this is dead code - good catch 👍

File: src/js/src/store/modules/ngram.store.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+  results: {},
```
Comment (jesperlauridsen) [inline src/js/src/store/modules/ngram.store.js:9]: state.results isn't really used for anything significant. You put the current result in it after a successfull search, but use the state.datasets for drawing the chart. The only real use, as far as i can see, is in SearchBox.vue where you use it to check if the "reset"-button should be shown - but you can just look at the state.dataset variable instead. I think it should be removed to nice up the code :-)

File: src/js/src/components/ToolboxComponents/NgramNetarchive.vue [hunk 6]
```diff
@@ -0,0 +1,37 @@
+<template>
+  <div class="linkGraphContainer">
+    <h2 class="toolboxHeadline">
+      Ngram Netarchive
+    </h2>
+    <div class="contentContainer">
+      <transition name="loading-overlay">
+        <loading-overlay />
+      </transition>
+      <notifications />
```
Comment (jesperlauridsen) [inline src/js/src/components/ToolboxComponents/NgramNetarchive.vue:10]: Why are we needing the notifications here again? Just curious...

File: src/js/src/components/ngrams/chartsCore/chartConfigs/chartOptionsCore.js (hunk 1 above)
Comment (jorntx) [reply src/js/src/components/ngrams/chartsCore/chartConfigs/chartOptionsCore.js:117]: :)

File: src/js/src/components/ngrams/chartsCore/chartEngines/LineChart.js (hunk 2 above)
Comment (jorntx) [reply src/js/src/components/ngrams/chartsCore/chartEngines/LineChart.js:21]: I have access to the chart object here in LineChart no problem (also stated in the docs) but if I tamper with _this.option_ here even as data/computed property(!!) all hell breaks loose on first search. After first search all is fine.

File: src/js/src/components/ngrams/netarchive/SearchBox.vue (hunk 4 above)
Comment (jorntx) [reply src/js/src/components/ngrams/netarchive/SearchBox.vue:37]: 👍

File: src/js/src/store/modules/ngram.store.js (hunk 5 above)
Comment (jorntx) [reply src/js/src/store/modules/ngram.store.js:9]: Another good catch!

File: src/js/src/components/ToolboxComponents/NgramNetarchive.vue (hunk 6 above)
Comment (jorntx) [reply src/js/src/components/ToolboxComponents/NgramNetarchive.vue:10]: You are on 🔥 today! Leftover from the refactor in toolbox we can rely on the global one:)

File: src/js/src/components/ngrams/chartsCore/chartEngines/LineChart.js (hunk 2 above)
Comment (jesperlauridsen) [reply src/js/src/components/ngrams/chartsCore/chartEngines/LineChart.js:21]: Hmmm. It sounds... intriguing! I hope you find your solution to this at some point :pray:

**Review by jesperlauridsen — APPROVED** [review]
Looks very good. Good to get this into the application aswell. Well done! The only few things I found are just dead code, that could be removed, I think. Tested the branch out too - apart from the error you're already aware off, it works very nicely.

**Pull request merged**